File: Data Structures & Algorithms/two-integer-sum/submission-0.java
class Solution {
    int[] indices = new int[2];

    public int[] twoSum(int[] nums, int target) {
        for (int x = 0; x < nums.length; x++)
        {
            for (int y = x+1; y < nums.length; y++)
            {
                if (nums[y] == target-nums[x])
                {
                    indices[0] = x;
                    indices[1] = y;
                    break;
                }
            }
        }
        return indices;
    }
}

File: Data Structures & Algorithms/find-target-in-rotated-sorted-array/submission-0.go
func search(nums []int, target int) int {
	left, right := 0, len(nums)-1

	for left <= right{
		middle := (left + right) / 2

		//cek middle jawaban or no
		if nums[middle] == target{
			return middle
		}

		//sortednya left
		if nums[middle] >= nums[left]{

			//cek target diantara sorted left gk
			if target >= nums[left] && target <= nums[middle]{
				right = middle - 1
			// target di sorted right	
			} else {
				left = middle + 1
			}

		//sortednya right
		} else if nums[middle] <= nums[right]{
			if target >= nums[middle] && target <= nums[right]{
				left = middle + 1
			} else {
				right = middle - 1
			}
		}
	}
	return - 1
}
